github/messages.go

```diff
@@ -133,13 +136,45 @@ func messageMAC(signature string) ([]byte, func() hash.Hash, error) {
 //     }
 //
 func ValidatePayload(r *http.Request, secretKey []byte) (payload []byte, err error) {
-	payload, err = ioutil.ReadAll(r.Body)
-	if err != nil {
-		return nil, err
+	var body []byte // Raw body that GitHub uses to calculate the signature.
+
+	switch ct := r.Header.Get("Content-Type"); ct {
+	case "application/json":
+		var err error
+		body, err = ioutil.ReadAll(r.Body)
```

Comment:
nit:

```go
if body, err = ioutil.ReadAll(r.Body); err != nil {
```

Reply:
Oh, OK.  I thought it was more idiomatic Go, but that's OK.  Feel free to switch it back... I don't feel that strongly about it.  Sorry for the switch... I forgot you preferred the other way.

Reply:
> I thought it was more idiomatic Go

As far as I know, both are valid idiomatic Go. Which one is better to use depends on the context/situation.

I'll leave it as is, I don't feel very strongly either.